libsolidity/analysis/DocStringAnalyser.cpp

```diff
@@ -36,21 +36,53 @@ using namespace solidity::frontend;
 namespace
 {
 
-void copyMissingTags(StructurallyDocumentedAnnotation& _target, set<CallableDeclaration const*> const& _baseFunctions)
+void copyMissingTags(set<CallableDeclaration const*> const& _baseFunctions, StructurallyDocumentedAnnotation& _target, CallableDeclaration const* _declaration = nullptr)
 {
 	if (_baseFunctions.size() != 1)
 		return;
 
 	auto& sourceDoc = dynamic_cast<StructurallyDocumentedAnnotation const&>((*_baseFunctions.begin())->annotation());
 
-	set<string> existingTags;
-
-	for (auto const& iterator: _target.docTags)
-		existingTags.insert(iterator.first);
-
-	for (auto const& [tag, content]: sourceDoc.docTags)
-		if (tag != "inheritdoc" && !existingTags.count(tag))
-			_target.docTags.emplace(tag, content);
+	for (auto it = sourceDoc.docTags.begin(); it != sourceDoc.docTags.end();)
+	{
+		string const& tag = it->first;
+		// Don't copy tag "inheritdoc" or already existing tags
+		if (tag == "inheritdoc" || _target.docTags.count(tag))
+		{
+			it++;
+			continue;
+		}
+
+		size_t n = 0;
+		// Iterate over all values of the current tag (it's a multimap)
+		for (auto next = sourceDoc.docTags.upper_bound(tag); it != next; it++, n++)
+		{
+			DocTag const& content = it->second;
+			DocTag newContent = content;
+			newContent.inherited = true;
+
+			// Update the parameter name for @return tags
```

Comment:
What about `@param`?

Reply:
well, this issue was about `@return`. As far as I know, `@param` can't be unnamed so they don't have the same problem.
Of course the name can change and in that case nothing will be inherited.